benchmark/transform_benchmark.cpp

```diff
@@ -96,6 +103,38 @@ BENCHMARK(BM_transform_in_place_slice)
     ->RangeMultiplier(2)
     ->Ranges({{1, 2 << 18}, {false, true}});
 
+static void BM_transform_in_place_transposed(benchmark::State &state) {
+  const auto nx = 100;
+  const auto ny = state.range(0);
+  const auto n = nx * ny;
+  const bool use_variances = state.range(1);
+  auto a = makeBenchmarkVariable(Dimensions{{Dim::Y, ny}, {Dim::X, nx}},
+                                 use_variances);
+  auto b = makeBenchmarkVariable(Dimensions{{Dim::X, nx}, {Dim::Y, ny}},
+                                 use_variances);
```

Comment:
Just so I understand correctly: By using the transpose we ensure that we hit a non-optimized branch in `transform`?

If that is correct, should we use as small as `nx` as possible, to avoid measuring inefficient cache access due to the transposes data access?

Reply:
> Just so I understand correctly: By using the transpose we ensure that we hit a non-optimized branch in transform?

Yes Since there is no way to construct a strided slice, I figured this is the easiest way to disable the optimisation.

> If that is correct, should we use as small as nx as possible, to avoid measuring inefficient cache access due to the transposes data access?

Good point! Do you think that 10 is small enough / too small?

Reply:
I'd use a small power of 2, and maybe small enough so it fits a cache line (or half a cache line)?

Reply:
Hm, I'll try 4 (my cachelines are 64 bytes long).

Reply:
![transposed_benchmark](https://user-images.githubusercontent.com/11393224/104185997-0bdd5380-5416-11eb-9edc-6d2805af4ed2.png)
No significant regression. But also not improved, as expected.